src/browser/findDOMNode.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2013-2014, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule findDOMNode
+ * @typechecks static-only
+ */
+
+"use strict";
+
+var ReactComponent = require('ReactComponent');
+var ReactInstanceMap = require('ReactInstanceMap');
+var ReactMount = require('ReactMount');
+
+var invariant = require('invariant');
+var isNode = require('isNode');
+
+/**
+ * Returns the DOM node rendered by this element.
+ *
+ * @param {ReactComponent|DOMElement} element
+ * @return {DOMElement} The root node of this element.
+ */
+function findDOMNode(componentOrElement) {
+  if (componentOrElement == null) {
+    return null;
+  }
+  if (isNode(componentOrElement)) {
+    return componentOrElement;
+  }
+  if (ReactInstanceMap.has(componentOrElement)) {
+    return ReactMount.getNodeFromInstance(componentOrElement);
+  }
+  invariant(
+    !(componentOrElement.render != null && typeof(componentOrElement.render) === 'function'),
```

Comment:
I kind of agree that I used to think of it as "if...then...throw". However, the invariant model has a long standing tradition at Facebook since the early PHP days. It's also similar to `assert` statements in several other languages.

It's nice an declarative in a TDD kind of way. It has grown on me. At the end of the day, it doesn't really matter which inversion you chose. Different people will have different mental models here.

That said, the worst kind is the double negative. To think this one through, you might have to do two conversions! Demorganizing is hard enough already. People (me) screw this up all the time. I'd go as far as to say that it's objectively worse to have double negatives like this.